clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/AmazonCloudFormationCachingAgent.java

```diff
@@ -39,28 +43,57 @@
 import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE;
 
 @Slf4j
-public class AmazonCloudFormationCachingAgent implements CachingAgent, AccountAware {
+public class AmazonCloudFormationCachingAgent implements CachingAgent, OnDemandAgent, AccountAware {
   private final AmazonClientProvider amazonClientProvider;
   private final NetflixAmazonCredentials account;
   private final String region;
+  private final OnDemandMetricsSupport metricsSupport;
 
   static final Set<AgentDataType> types = new HashSet<>(
     Collections.singletonList(AUTHORITATIVE.forType(STACKS.getNs()))
   );
 
   public AmazonCloudFormationCachingAgent(AmazonClientProvider amazonClientProvider,
                                           NetflixAmazonCredentials account,
-                                          String region) {
+                                          String region,
+                                          Registry registry) {
     this.amazonClientProvider = amazonClientProvider;
     this.account = account;
     this.region = region;
+    this.metricsSupport = new OnDemandMetricsSupport(registry, this,
+      String.format("%s:%s", AmazonCloudProvider.ID, OnDemandType.CloudFormation));
   }
 
   @Override
   public String getProviderName() {
     return AwsInfrastructureProvider.PROVIDER_NAME;
   }
 
+  @Override
+  public String getOnDemandAgentType() {
+    return getAgentType();
+  }
+
+  @Override
+  public OnDemandMetricsSupport getMetricsSupport() {
+    return this.metricsSupport;
+  }
+
+  @Override
+  public boolean handles(OnDemandType type, String cloudProvider) {
+    return type == OnDemandType.CloudFormation && cloudProvider.equals(AmazonCloudProvider.ID);
```

Comment:
I think we'd prefer to use `OnDemandType.CloudFormation.equals(type)` instead of `==`.

Reply:
Done in https://github.com/spinnaker/clouddriver/pull/3295/commits/36580ea576634912796418ed75b2578806d782e9